File: modified_scripts/julia/merged_scripts/merged_is_palindrome.jl

# qwen3_1_7b CODE
function is_palindrome(s::String)
    return s == reverse(s)
end
# qwen2_5-coder_7b CODE

function is_palindrome(s::String)::Bool
    n = length(s)
    for i in 1:div(n,2)
        if s[i] != s[n-i+1]
            return false
        end
    end
    return true
end

# qwen3_8b CODE
function is_palindrome(s)
    i = 1
    j = length(s)
    while i < j
        if s[i] != s[j]
            return false
        end
        i += 1
        j -= 1
    end
    return true
end
# gemma_2b CODE
function is_palindrome(string)
    n = length(string)
    center = floor(n/2)
    for i = 1:n
        if string[i] != string[n - i]
            return(false)
    end
    return(true)
end

# deepseek-coder_1_3b CODE

function is_palindrome(s)
    length = div(length(s),2 ) 
    
    i = 1           
                     
    j = length       
    
    
    t = ""             
    
    while i <= j
        if s[i] != s[j]   
            return false    
        else                
            i += 1          
            j -= 1          
        end               
     end                

# gemma_7b CODE
function is_palindrome(str)
    str = lower(strip(str))
    return str == rev(str)
end

# deepseek-coder_6_7b CODE

function is_palindrome(s::String)
    len = length(s)
    for i = 1:len ÷ 2
        if s[i] != s[len - i + 1]
            return false
        end
    end
    return true
end

# qwen2_5-coder_3b CODE

function is_palindrome(s::String) = s == reverse(s)

File: modified_scripts/rust/merged_scripts/merged_is_palindrome.rs

# gemma_7b CODE

fn is_palindrome(string: &str) -> bool {
    string.eq(string.chars().rev().collect::<String>())
}

# deepseek-coder_6_7b CODE
pub fn is_palindrome(input: &str) -> bool {
    let input = input.to_lowercase();
    let len = input.len();
    for i in 0..len/2 {
        if input.chars().nth(i) != input.chars().nth(len-1-i) {
            return false;
        }
    }
    true
}

# qwen2_5-coder_3b CODE

fn is_palindrome(s: &str) -> bool {
    let chars: Vec<char> = s.chars().collect();
    chars.iter().eq(chars.iter().rev())
}

# deepseek-coder_1_3b CODE
pub fn is_palindrome(input : &str) -> bool {
    let input = String::from(input); 
    
   if !input.is_empty() && (&mut '0'.to_string()) == &rev(&input) {} else { false }     
}   

# qwen3_1_7b CODE
fn is_palindrome(s: &str) -> bool {
    let mut chars: Vec<char> = s.chars().collect();
    let mut left = 0;
    let mut right = chars.len() - 1;

    while left < right {
        if chars[left] != chars[right] {
            return false;
        }
        left += 1;
        right -= 1;
    }
    true
}
# qwen2_5-coder_7b CODE

fn is_palindrome(s: &str) -> bool {
    let mut chars = s.chars();
    while let (Some(f), Some(l)) = (chars.next(), chars.next_back()) {
        if f != l {
            return false;
        }
    }
    true
}

# qwen3_8b CODE
fn is_palindrome(s: &str) -> bool {
    let mut left = 0;
    let mut right = s.len() - 1;
    while left < right {
        if s.as_bytes()[left] != s.as_bytes()[right] {
            return false;
        }
        left += 1;
        right -= 1;
    }
    true
}
# gemma_2b CODE

fn is_palindrome(text: &str) -> bool {
    text.chars().rev().eq(text.chars());
}
